### ISSUE TITLE
Line too long error when using hsts annotations

### ISSUE THREAD

**Issue by abh** [jcmoraisjr/haproxy-ingress#586]
I added some of the hsts annotations to one of our ingresses (in a namespace with a lot of ingress domains) and it causes haproxy to be unable to reload the config, failing with this error

```
[ALERT] 138/225325 (176) : parsing [/etc/haproxy/haproxy.cfg:746]: line too long, truncating at word 65, position 493: <path68 path73 path74 path38 path36 path54 path25 path62 path46 path42 path71 path78 path79 path80 path40 path85 path84 path57 path11 path10 path33 path31 path32 path60 path50 path18 path37 path72 path35 }>.
```

(might be related to #361)

**Comment by jcmoraisjr** [comment]
Hi, this is the same problem but with orders of magnitude greater. It sounds to me that the problem isn't the number of domains, but instead the number of distinct paths on a single domain and you partially configured hsts into that domain. Please confirm if:

* you have at least one domain with 70 distinct paths or so;
* you don't have the same hsts configuration applied to all of the paths;
* the problem disapear when you configure exactly the same hsts configuration on all paths of this domain

**Comment by daklik** [comment]
Hi, facing quite the same issue with v0.9 - where we had no problem with v0.7.2.
We have like 60 domains listed in our ingress config, each with the same / path. 
We have also a basic-auth on top of this ingress.
The error is on this http-auth section: 
```
[ALERT] 294/073733 (40) : parsing [/etc/haproxy/haproxy.cfg:295]: line too long, truncating at word 65, position 459: <path56 } !{ http_auth(web-experience-mydoma │
│ [ALERT] 294/073733 (40) : parsing [/etc/haproxy/haproxy.cfg:295] : error detected while parsing an 'http-request auth' condition : missing closing '}' in condition │
│ [ALERT] 294/073733 (40) : Error(s) found in configuration file : /etc/haproxy/haproxy.cfg 
```

**Comment by jcmoraisjr** [comment]
I've just figured out that you both should be sharing the same backend (service+port) with 50+ distinct hostnames, most of them have custom configs (respectively hsts and basic auth), and these configs differ on at least one hostname or path. v0.7 and older doesn't have this problem because such triage happened in a per hostname/path basis, which was quite expensive to haproxy. Thanks for the insight, I'm taking care of this one.

**Comment by jcmoraisjr** [comment]
Fixed in v0.11-beta.4 and already merged to v0.10 as well, new release in a week or so. Closing.

**Comment by abh** [comment]
thank you!!
